### ISSUE TITLE
android unauthorized state detection is broken

### ISSUE THREAD

**Issue by vdelendik** [zebrunner/appium#361]
@azarouski, you have to repro negative cases to make sure you read negative states as well.
below is example of the unauthorized state which can't be detected by new appium

```
✔ Listing available plugins
- images [not installed]
- execute-driver [not installed]
- relaxed-caps [not installed]
- universal-xml [not installed]
03-15 14:01:04.947    94    94 I adb     : auth.cpp:416 adb_auth_init...
03-15 14:01:04.953    94    94 I adb     : auth.cpp:152 loaded new key from '/root/.android/adbkey' with fingerprint A6530430B5C6D04750CFB681BB018198D8395A1F63ACE92C86261449FDED532C
03-15 14:01:04.953    94    94 I adb     : auth.cpp:391 adb_auth_inotify_init...
03-15 14:01:04.953    94   102 I adb     : transport.cpp:307 1322420048457: write thread spawning
03-15 14:01:04.953    94   101 I adb     : transport.cpp:335 1322420048457: read thread spawning
03-15 14:01:04.957    94    94 I adb     : transport.cpp:1734 fetching keys for transport 1322420048457
03-15 14:01:04.957    94    94 I adb     : auth.cpp:468 Calling send_auth_response
03-15 14:01:04.964    94    94 I adb     : auth.cpp:435 Calling send_auth_publickey
error: device unauthorized.
This adb server's $ADB_VENDOR_KEYS is not set
Try 'adb kill-server' if that seems wrong.
Otherwise check for a confirmation dialog on your device.
Not documented device state: ''. One more attempt in 5 seconds.
Restarting...
Doing usbreset forcibly on attached device
Resetting USB device /dev/bus/usb/002/093
Reset successful
```

make sure to do exit 0 asap and don't restart

**Comment by vdelendik** [comment]
device state detection fix was in registering stderr in the state variable here:
https://github.com/zebrunner/appium/blob/1c73e8e5887d19ff7f506fff745e34fe2d3076b7/device_connect.sh#L86
```
state=$(adb get-state)
->
state=$(adb get-state 2>&1)
```

also for stderr we had to verify using contains logic like `*"authorizing"*` inside case operator
https://github.com/zebrunner/appium/blob/1c73e8e5887d19ff7f506fff745e34fe2d3076b7/device_connect.sh#L100

changes are available for testing in `appium:2.0.14` because I already applied changes on one of the prod as a hot fix

**Comment by vdelendik** [comment]
tested successfully on my own:
```state: error: device unauthorized. This adb server's $ADB_VENDOR_KEYS is not set Try 'adb kill-server' if that seems wrong. Otherwise check for a confirmation dialog on your device.

Authorize device manually!
Exiting without restarting...
```
